src/ctapipe/monitoring/outlier.py

```diff
@@ -81,7 +81,7 @@ class MedianOutlierDetector(OutlierDetector):
     the configurable factors and the camera median of the statistic values.
     """
 
-    median_range_factors = List(
+    validity_range = List(
```

Comment:
Why this change?

Reply:
We would need the same generic name in each `OutlierDetector` class in order to init them homogeneously in the `PixelStatisticsCalculator`.

Reply:
And what if someone comes up with an `OutlierDetector` that works differently?

Reply:
I also do not understand this argument. You have a full dict config for each of the outlier detectors.

What prevents you from configuring:

```
{
  "outlier_detector_list": [
    {
      "apply_to": "mean",
      "name": "MedianOutlierDetector",
      "median_range_factor": [-2.0, 2.0],
    },
    {
      "apply_to": "mean",
      "name": "StdOutlierDetector",
      "std_range_factor": [-2.0, 2.0],
    },
  ],
}
```

or maybe even better:

```
{
  "outlier_detector_list": [
    {
      "apply_to": "mean",
      "name": "MedianOutlierDetector",
      "config": {
        "median_range_factor": [-2.0, 2.0],
      }
    },
    {
      "apply_to": "mean",
      "name": "StdOutlierDetector",
      "config": {
        "std_range_factor": [-2.0, 2.0],
      },
    },
  ],
}
```

Reply:
Using the first example, you could do (from memory, not tested...):

```
for outlier_detector in enumerate(self.outlier_detector_list):
    # copy, so we don't mutate the actual configuration
    outlier_detector = outlier_detector.copy()
    cls_name = outlier_detector.pop("name")
    apply_to = outlier_detector.pop("apply_to")
    self.outlier_detectors.append(OutlierDetector.from_name(cls_name, parent=self, **outlier_detector)
```

Reply:
For me having the name to appear in the class name and in the config variable is a self-repetition. 
We have an outlier detector, that uses a validity range, this is a generic concept, which is specialized in the derived outlier detectors. Now, if someone develops an outlier detector that doesn't follow this logic, it should not be derived from the same parent (and btw, what would be a use case for this? I can't figure that out).

Reply:
Before, it made clear that it is a relative factor, not an absolute range.

> what would be a use case for this? 

There are many difference outlier detection algorithms. Not all of them have a two parameter valid range. 
One example would be sigma clipping.

In general, we don't require that subclasses of components have the same configuration options.

Quite the opposite actually. The should have the same API, so they can be used interchangebly in the code, but the configuration is specific to each one and the configuration system takes care of that.

This is the main use case of the config / component systems: have classes that have the same API in the code, but that can be configured using different options. See e.g. the cleaning or image extraction components.

Reply:
Ah, I see your point better now. In my opinion, the configuration parameter meaning shall be properly documented in the specific implementation of the OutlierDetector. Also, we do not require two parameter valid range. Perhaps a better name can work, to have it generic (with a proper description in the specific implementation). But I don't have a strong opinion on that.

Reply:
Above I posted a code snipped that would allow passing any configuration option to the specific outlier detector. 

Independently of what is the best name for the options of the current detectors, I think this is the way to go.

I would prefer if this PR wasn't making API breaking changes in code that is not directly affected by the main goal of this PR.